src/errors.ts

```diff
@@ -1,131 +1,32 @@
-    super(message, code, true, options);
-  }
-}
-
-export class HttpError extends RequestError {
-  statusCode: number;
-  extensions?: ErrorExtensions;
-  response: ErrorResponse;
-  headers: IncomingHttpHeaders;
-
-  constructor(
-    message: string,
-    statusText: string,
-    statusCode: number,
-    extensions: ErrorExtensions | undefined,
-    response: ErrorResponse,
-    headers: IncomingHttpHeaders,
-    options?: ErrorOptions,
-  ) {
-    super(message, 'ERR_NON_2XX_3XX_RESPONSE', false, options);
-    this.statusCode = statusCode;
-    this.extensions = extensions;
-    this.response = response;
-    this.headers = headers;
-  }
-}
-
-function isAbortError(err: unknown): err is DOMException {
-  return Boolean(err && err instanceof Error && err.name === 'AbortError');
-}
-
-export function errorTransformer(err: unknown) {
-  if (isAbortError(err)) {
-    return new RequestCanceledError(
-      err.message,
-      String(err.cause ?? err.name),
-      {
-        cause: err,
-      },
-    );
-  }
-
-  if (!isAxiosError(err)) {
-    return err;
-  }
-
-  if (err.response) {
-    // The request was made and the server responded with a status code
-    // that falls out of the range of 2xx
-    const response = err.response as AxiosResponse<ErrorResponse>;
-
-    return new HttpError(
-      response.data.message,
-      response.statusText,
-      response.status,
-      response.data.extensions,
-      response.data,
-      (response.headers instanceof AxiosHeaders
-        ? response.headers.toJSON()
-        : // There's some inconsistency between node headers and axios headers types (and only types because we are not in the browser)
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          response.headers) as any,
-      { cause: err },
-    );
-  }
-
-  if (isAxiosCancel(err as unknown)) {
-    return new RequestCanceledError(err.message, err.code, { cause: err });
-  }
-  return new RequestError(err.message, err.code, false, { cause: err });
-}
-
-export function isRetrievableError(error: unknown): boolean {
-  if (error instanceof HttpError) {
-    return [
-      // Client errors
-      HttpStatusCode.RequestTimeout,
-      HttpStatusCode.TooManyRequests,
-      // Server errors
-      HttpStatusCode.InternalServerError,
-      HttpStatusCode.BadGateway,
-      HttpStatusCode.ServiceUnavailable,
-      HttpStatusCode.GatewayTimeout,
-      HttpStatusCode.InsufficientStorage,
-    ].includes(error.statusCode);
-  }
-  if (error instanceof RequestError) {
-    return !error.cancelled;
+export abstract class RequestError extends BaseError {}
+
+export class NetworkError extends RequestError {}
+
+export class HttpError extends RequestError implements ApiError {
+  readonly error: string;
+  readonly status_code: number;
+  readonly extensions: {
+    code: string;
+    state?:
+      | {
+          [key: string]: unknown;
+        }
+      | null
+      | undefined;
```

Comment:
`undefined` is not needed

Reply:
`state` is always defined -> `Record<string, unknown>`

Reply:
Switched to generated types but note that the type is indeed ```{
            [key: string]: unknown;
        } | null | undefined;```